satpy/utils.py

```diff
@@ -538,3 +539,86 @@ def get_chunk_pixel_size():
     else:
         array_size = CHUNK_SIZE ** 2
     return array_size
+
+
+def convert_remote_files_to_fsspec(filenames, storage_options=None):
+    """Check filenames for transfer protocols, convert to FSFile objects if possible."""
+    if storage_options is None:
+        storage_options = {}
+    if isinstance(filenames, dict):
+        return _check_file_protocols_for_dicts(filenames, storage_options)
+    return _check_file_protocols(filenames, storage_options)
+
+
+def _check_file_protocols_for_dicts(filenames, storage_options):
+    res = {}
+    for reader, files in filenames.items():
+        opts = storage_options.get(reader, {})
+        res[reader] = _check_file_protocols(files, opts)
+    return res
+
+
+def _check_file_protocols(filenames, storage_options):
+    local_files, remote_files, fs_files = _sort_files_to_local_remote_and_fsfiles(filenames)
+
+    if remote_files:
+        return local_files + fs_files + _filenames_to_fsfile(remote_files, storage_options)
+
+    return local_files + fs_files
+
+
+def _sort_files_to_local_remote_and_fsfiles(filenames):
+    from urllib.parse import urlparse
+
+    from satpy.readers import FSFile
```

Comment:
Both of these imports are in satpy or standard library so these can be moved to the top of the module right?

Reply:
Oh, indeed. I was going to argue `fsspec` would then be a hard requirement, but `FSFile` doesn't import it. Which reminds me I need to add `fsspec` as optional dependency to `setup.py`...

Reply:
Done and done.

Reply:
Needed to move `FSFile` import back to the function due to circular import.